app/assets/javascripts/discourse/app/components/pick-files-button.js

```diff
@@ -0,0 +1,105 @@
+import Component from "@ember/component";
+import { action } from "@ember/object";
+import { empty } from "@ember/object/computed";
+import { default as computed } from "discourse-common/utils/decorators";
+import I18n from "I18n";
+
+export default Component.extend({
+  classNames: ["pick-files-button"],
+  acceptedFileTypes: null,
+  acceptAnyFile: empty("acceptedFileTypes"),
+
+  didInsertElement() {
+    this._super(...arguments);
+    this._bindControls();
+  },
+
+  @computed
+  acceptedFileTypesString() {
+    if (!this.acceptedFileTypes) {
+      return null;
+    }
+
+    return this.acceptedFileTypes.join(",");
+  },
+
+  @computed
+  acceptedExtensions() {
+    if (!this.acceptedFileTypes) {
+      return null;
+    }
+
+    return this.acceptedFileTypes
+      .filter((type) => type.startsWith("."))
+      .map((type) => type.substring(1));
+  },
+
+  @computed
+  acceptedMimeTypes() {
+    if (!this.acceptedFileTypes) {
+      return null;
+    }
+
+    return this.acceptedFileTypes.filter((type) => !type.startsWith("."));
+  },
+
+  @action
+  openSystemFilePicker() {
+    document.querySelector("#file-input").click();
```

Comment:
We should use `this.element.querySelector` to scope it to this component rather than the whole document. Also since we are finding by Id I recommend `this.element.getElementById`. (same for `_bindControls` below)

Reply:
Turned out `getElementById` isn't available on `element`, only on `document`. Anyway, I think we don't need id on the input, we can select it using `this.element.querySelector("input")`. I've addressed this and all other comments.